### PR TITLE
SOF-1374 Live cuts directly

### PR DESCRIPTION
**Pull request by LindvedKrvang**
Able to create ActionTriggers for Live that cuts directly to program.

This is basically extracting the code from `executeActionCutToCamera` that determines when and how to cut directly and make `executeActionCutToRemote` use the extracted code.

Naming suggestions for `executePiece` is much appreciated. If you want me to refactor `executePiece` further, let me know (currently just extracted it and changed some variable names)

### CODE REVIEW

**Review by ianshade — APPROVED** [review]
Besides two minor comments it looks very good!
Since the shared `executePiece` can now support both queued and immediately taken pieces for Remotes and Cameras, I've been thinking that maybe we could get rid of `makeRemoteAdLibs` method in `GlobalAdLibPiecesGenerator` and use actions for queued Remotes too. And `EvaluateEksternBase` could create an adlib action as well. But it's up to you whether it shouldbedone within this PR or not

File: src/tv2-common/actions/actionTypes.ts [hunk 1]
```diff
@@ -60,6 +60,7 @@ export interface ActionCutToCamera extends ActionBase {
 
 export interface ActionCutToRemote extends ActionBase {
 	type: AdlibActionType.CUT_TO_REMOTE
+	cutDirectly: boolean
```
[the remark this answers is not in the document]
Comment (LindvedKrvang) [reply src/tv2-common/actions/actionTypes.ts:63]: Correct, it does the exact "opposite" of `queue` in `ActionCutToCamera`.
The reason for making it "opposite" was so the default behaviour of `ActionCutToRemote` would be the same as before this change.

By aligning them do you mean combining `ActionCutToCamera` and `ActionCutToRemote` into one interface or just have `cutDirectly` and `queue` be the same on both interfaces? Or maybe something third? I'm up for us making some further changes to this :)

File: src/tv2-common/actions/executeAction.ts [hunk 2]
```diff
@@ -1186,12 +1201,7 @@ async function executeActionCutToRemote<
 		}
 	}
 
-	await context.core.queuePart(part, [
-		remotePiece,
-		...(settings.SelectedAdlibs
-			? await getPiecesToPreserve(context, settings.SelectedAdlibs.SELECTED_ADLIB_LAYERS, [])
-			: [])
-	])
+	await executePiece(context, settings, remotePiece, !userData.cutDirectly, part)
```
[the remark this answers is not in the document]
Comment (LindvedKrvang) [reply src/tv2-common/actions/executeAction.ts:1204]: I will investigate

File: src/tv2-common/actions/executeAction.ts [hunk 3]
```diff
@@ -1053,34 +1045,57 @@ async function executeActionCutToCamera<
 		}
 	}
 
-	if (userData.queue || serverInCurrentPart) {
-		await context.core.queuePart(part, [
-			kamPiece,
+	await executePiece(context, settings, kamPiece, userData.queue, part)
```
Comment (RasmusAlbrektsen) [inline src/tv2-common/actions/executeAction.ts:1048]: I find it so confusing when we switch languages on things like `kamPiece`, but i know it's just taken from elsewhere.

**Review by RasmusAlbrektsen — APPROVED** [review]
Looks good from what i can see :-)

File: src/tv2-common/actions/executeAction.ts [hunk 4]
```diff
@@ -1053,34 +1045,57 @@ async function executeActionCutToCamera<
...
+	ShowStyleConfig extends TV2BlueprintConfigBase<StudioConfig>
+>(
+	context: ActionExecutionContext<ShowStyleConfig>,
+	settings: ActionExecutionSettings<StudioConfig, ShowStyleConfig>,
+	pieceToExecute: IBlueprintPiece<PieceMetaData>,
+	shouldBeQueued: boolean,
+	partToQueue: IBlueprintPart
+) {
+	const currentPieceInstances = await context.core.getPieceInstances('current')
+	const serverInCurrentPart = currentPieceInstances.some(
```
Comment (KvelaGorrrrnio) [inline src/tv2-common/actions/executeAction.ts:1062]: When I read this I expect the `serverInCurrentPart` to be a Part describing a server clip, but the value is a boolean. I'll suggest renaming it to `isServerInCurrentPart`.

**Review by KvelaGorrrrnio — APPROVED** [review]

File: src/tv2-common/actions/executeAction.ts (hunk 4 above)
Comment (LindvedKrvang) [reply src/tv2-common/actions/executeAction.ts]: Updated

File: src/tv2-common/actions/executeAction.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (LindvedKrvang) [reply src/tv2-common/actions/executeAction.ts:1205]: Cutting directly to a live now also has ´isPieceInjectedInPart´ set to true. As far as my testing goes, it appears cutting directly to a live has the same behaviour as cutting directly to a camera

**Pull request merged**